sympy/physics/quantum/tensorproduct.py

```diff
@@ -141,10 +141,8 @@ def flatten(cls, args):
     def _eval_adjoint(self):
         return TensorProduct(*[Dagger(i) for i in self.args])
 
-    def _eval_rewrite(self, pattern, rule, **hints):
-        sargs = self.args
-        terms = [t._eval_rewrite(pattern, rule, **hints) for t in sargs]
-        return TensorProduct(*terms).expand(tensorproduct=True)
+    def _eval_rewrite(self, rule, args, **hints):
+        return TensorProduct(*args).expand(tensorproduct=True)
```

Comment:
Can we just get rid of this? This should really just be expand or something rather than rewrite.

Reply:
That should be done in another PR, I suppose.